src/tracing.rs

```diff
@@ -0,0 +1,29 @@
+use anyhow::Result;
+use tracing_subscriber::filter::LevelFilter;
+use tracing_subscriber::fmt::time::ChronoLocal;
+use tracing_subscriber::FmtSubscriber;
+
+pub fn init(level: LevelFilter, json_format: bool) -> Result<()> {
+    if level == LevelFilter::OFF {
+        return Ok(());
+    }
+
+    let is_terminal = atty::is(atty::Stream::Stderr);
+
+    let builder = FmtSubscriber::builder()
+        .with_env_filter(format!("rendezvous_server={}", level))
+        .with_writer(std::io::stderr)
+        .with_ansi(is_terminal)
+        .with_timer(ChronoLocal::with_format("%F %T".to_owned()))
+        .with_target(false);
+
+    if json_format {
+        builder.json().init();
+    } else if is_terminal {
+        builder.init();
+    } else {
+        builder.without_time().init();
+    }
+
+    Ok(())
```

Comment:
Not a fan of such `if-else` chains. I'd say the following is easier to reason about:

```suggestion
   if json_format {
   		builder.json().init();
   		return Ok(())
   }
   
   if !is_terminal {
   		builder.without_time().init();
   		return Ok(())
   }

	builder.init();
	
	Ok(())
```

Reply:
I am aware that I made the original proposal in xmr-btc but in the spirit of what I said this morning in the retro, I think doing "magic" like not displaying time in the logs because we are not a terminal is not particularly good.

I'd suggest we either remove this feature or make it toggle-able based on a flag the user can provide.

The original idea of this `without_time` was that when logs are redirected to `journald`, a timestamp is already recorded so we don't need to send one on stdout.